### PR TITLE
fix: Quickly clicking do it for me button shows errors during onboarding

### LINKED ISSUE #4808: [Bug]The Welcome Tour fails->The error pop up when the user clicks the Lets go button quickly
## Description

[What happened ] In Welcome tour feature->When the user click on Do it for me button continuously then the error is displayed in widget and we are not able to continue the tour

Steps to reproduce the behaviour:

1.In the application page
2.click on Welcome tour
3.Click on Lets go Button
4.Click on Start Building button
5.Clik on do it for me button continuously
6.Validate the Functionality

expected result:
In Welcome tour feature->When the user click on Do it for me button continuously there should be pause or button should be disabled until the action is completed then allow to click on Lets go button so there should not be error
Actual Result:
In Welcome tour feature->When the user click on Do it for me button continuously then the error is displayed in widget and we are not able to continue the tour

**Attached Video**
https://drive.google.com/file/d/1R1PCrTERBWUofOVtNaZ3yjfpQN-fZkDp/view

### Important Details

- Version: Cloud 
- OS: Windows
- Browser Chrome
- Environment Production

### PR DESCRIPTION
**Pull request by akash-codemonk**
## Description

Fix issue where quickly clicking the do it for me button attempts to create multiple input widgets with the same name in onboarding

Fixes #4808

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Manually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/input-widget-step 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.85 **(-0.01)** | 36.61 **(-0.01)** | 34.17 **(0)** | 55.38 **(-0.01)**
 :red_circle: | app/client/src/components/editorComponents/Onboarding/Helper.tsx | 39.77 **(-3.44)** | 33.63 **(-0.6)** | 0 **(0)** | 43.21 **(-3.46)**
 :red_circle: | app/client/src/sagas/OnboardingSagas.ts | 17.22 **(-0.17)** | 1.33 **(-0.06)** | 11.11 **(0)** | 17.61 **(-0.11)**</details>

### CODE REVIEW

File: app/client/src/components/editorComponents/Onboarding/Helper.tsx [hunk 1]
```diff
@@ -353,7 +364,7 @@ function Helper() {
             <CheatActionButton
               className="t--onboarding-cheat-action"
               onClick={() => {
-                dispatch(helperConfig.cheatAction?.action);
+                debouncedOnClick(helperConfig.cheatAction?.action);
```
Comment (rishabhsaxena) [inline app/client/src/components/editorComponents/Onboarding/Helper.tsx:367]: we could also use a flag using a `ref` here?

**Review by rishabhsaxena — DISMISSED** [review]

File: app/client/src/components/editorComponents/Onboarding/Helper.tsx (hunk 1 above)
Comment (akash-codemonk) [reply app/client/src/components/editorComponents/Onboarding/Helper.tsx:367]: Sorry i didn't get what you meant

Comment (rishabhsaxena) [reply app/client/src/components/editorComponents/Onboarding/Helper.tsx:367]: so, I meant added a flag and setting it to true on the first click, for example:
```
const isClickedRef = useRef(false);

const onClick = () => {
 if (isClickedRef.current) return;
 isClickedRef.current = true;
 ...
}
```

Comment (akash-codemonk) [reply app/client/src/components/editorComponents/Onboarding/Helper.tsx:367]: Nope this wouldn't work as it doesn't get unmounted between steps. The animation is just shown on top of it.

Comment (rishabhsaxena) [reply app/client/src/components/editorComponents/Onboarding/Helper.tsx:367]: we could reset it when the step changes? along with resetting the `cheatMode`?

**Review by rishabhsaxena — APPROVED** [review]

**Pull request merged**